### DISCUSSION TITLE
Libcamera.so.0.0.4: cannot open shared object file

### DISCUSSION THREAD

**Discussion by oave** [Motion-Project/motionplus#83] [Q&A]
Hi 

I have Motionplus installed on a raspberrypi 3B+ (bullseye). Installed via .deb (bullseye_motionplus_0.1.1-1_arm64.deb)
Since yesterday motionplus has stopped working, it stops before any log is saved. 
What i have tried: 
`$ sudo service motionplus start
$ sudo service motionplus status`

I got:
` motionplus.service - MotionPlus - Enhanced security camera monitoring software.
     Loaded: loaded (/lib/systemd/system/motionplus.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Wed 2023-07-19 15:35:08 CEST; 3s ago
       Docs: man:motionplus(1)
    Process: 5220 ExecStartPre=/bin/mkdir -p /var/log/motionplus (code=exited, status=0/SUCCESS)
    Process: 5221 ExecStartPre=/bin/chown motion:adm /var/log/motionplus (code=exited, status=0/SUCCESS)
    Process: 5222 ExecStart=/usr/bin/motionplus (code=exited, status=127)
   Main PID: 5222 (code=exited, status=127)
        CPU: 55ms`

if I try:
`$ libcamerify motionplus`

I got
`motionplus: error while loading shared libraries: libcamera.so.0.0.4: cannot open shared object file: No such file or directory`

Indeed `libcamera.so.0.0.4` does not exist, instead I found `libcamera.so.0.0.5`. As it was working perfectly last week I guess that libcamera got updated and replace  libcamera.so.0.0.4 by libcamera.so.0.0.5. 

I thought that creating symbolic links could solve the problem but it doesn't. It bypass the "not such file or directory" issue but Motionplus still does not run correctly.

**Comment by gcoltman** [comment]
Not sure if 'Me too's are allowed on Github but me too! Any helpful suggestions?

**Comment by Mr-Dave** [comment]
You'll need to rebuild Motionplus from source due to the change upstream on libcamera.

**Comment by gcoltman** [comment]
OK, thanks Mr-Dave!

**Comment by Mr-Dave** [comment]
From the associated [issue](https://github.com/Motion-Project/motionplus/issues/84#issuecomment-1661917209) from @rhk:

> Bumped into the same on Raspberry, first time trying Motionplus (have been using motion for quite some years)
rhk@mycabin:~ $ lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description: Raspbian GNU/Linux 11 (bullseye)
Release: 11
Codename: bullseye

>Installed pi_bullseye_motionplus_0.1.1-1_armhf.deb from website.

>rhk@mycabin:~ $ motionplus
motionplus: error while loading shared libraries: libcamera.so.0.0.4: cannot open shared object file: No such file or directory

>(didn't now know how to check libcamera version)

>Could someone elaborate, what should a 'normal' user do? Wait for a new build of motionplus to appear on the website or wait for motionplus to be packaged for debian/raspbian - and use motion while waiting?


Building your own deb from source is the option right now as described [here](https://motion-project.github.io/motionplus_build.html#Build_Deb).  

EDIT 2023/08/17:   Updated deb files are now on the [releases](https://github.com/Motion-Project/motionplus/releases/tag/release-0.1.1) page.
